### ISSUE TITLE
S3 backend insist to use bucket in role region regardless the region variable.

### ISSUE THREAD

**Issue by Magicloud** [hashicorp/terraform#23787]
Terraform 0.12.18.

I created a S3 bucket in ap-northeast-1 region. Then I use it by code:

```HCL
terraform {
  backend "s3" {
    encrypt = true
    bucket  = "eks-remote-state"
    key     = "mod/remote-state-storage/terraform.tfstate"
    region  = "ap-northeast-1"
  }
}
```

Terraform failed with message:

```
Error: Error loading state:
    BucketRegionError: incorrect region, the bucket is not in 'us-east-2' region at endpoint ''
	status code: 301, request id: , host id: 

Terraform failed to load the default state from the "s3" backend.
```

us-east-2 is the region of my AWS profile. I have to update my AWS profile to get the code working.

**Comment by UnixMonky** [comment]
I also have this same issue.  I've created the basic backend configuration (s3 bucket and DynamoDB) table in us-east-1. Attempting to access that s3 bucket when the default provider{} has a different region defined gives me the exact same error, despite having the region set in the configuration.

I did attempt to set a secondary provider and use the "alias" option, but that also failed; I suspect that the backend definition does not understand the alias option.

My configs for reference:
`provider "aws" {
  region                  = "us-east-2"
}


###
# Backend state store configuration
###
terraform {
  backend "s3" {
    bucket                  = "terraform-6K5KHG5BK-state"
    key                     = "backend/terraform.tfstate"
    region                  = "us-east-1"
    dynamodb_table          = "terraform-state-lock"
  }
}
`

**Comment by gdavison** [comment]
This will be addressed in Terraform v1.6. If you still see this error when v1.6 is released (or you try any of the v1.6 beta releases), please open a new issue
